pilot/pkg/model/push_context_test.go

```diff
@@ -229,6 +233,84 @@ func TestEnvoyFilters(t *testing.T) {
 	}
 }
 
+func TestInitPushContext(t *testing.T) {
+	env := &Environment{}
+	configStore := NewFakeStore()
+	configStore.Create(config.Config{
+		Meta: config.Meta{
+			Name:             "rule1",
+			Namespace:        "test1",
+			GroupVersionKind: gvk.VirtualService,
+		},
+		Spec: &networking.VirtualService{
+			Hosts:    []string{"rule1.com"},
+			ExportTo: []string{".", "ns1"},
+		},
+	})
+	configStore.Create(config.Config{
+		Meta: config.Meta{
+			Name:             "rule1",
+			Namespace:        "test1",
+			GroupVersionKind: gvk.DestinationRule,
+		},
+		Spec: &networking.DestinationRule{
+			ExportTo: []string{".", "ns1"},
+		},
+	})
+	store := istioConfigStore{ConfigStore: configStore}
+
+	env.IstioConfigStore = &store
+	env.ServiceDiscovery = &localServiceDiscovery{
```

Comment:
Can we include some tests with ExportTo set? 

I was wondering if it's problematic the initial indexing (by port, hostname, hostnameandnamespace) when a service has ExportTo Public? 

https://github.com/istio/istio/blob/419757834026f726fdaac8e52363ded25e7cefd8/pilot/pkg/model/push_context.go#L1067-L1069

Reply:
I am not sure I fully captured what you were thinking, but I added a service with exportTo.

Reply:
My guess is that svc2 will be missing from both in the "by port" instance in this test. I'm assuming the `By*` are supposed to be populated no matter the ExportTo value.

Reply:
I see, so the test passes since its checking a diff. You are expecting the fresh init is broken, not the copy? I will investigate

Reply:
Ah I see. I think you are right

Reply:
Great catch - should be fixed and tested now